x-pack/plugins/aiops/public/components/log_categorization/format_category.test.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import type { Category } from '../../../common/api/log_categorization/types';
+import { FormattedPatternExamples, FormattedRegex, FormattedTokens } from './format_category';
+
+const categories = [
+  {
+    key: 'Processed messages out of',
+    examples: ['Processed 676 messages out of 676'],
+    regex: '.*?Processed.+?messages.+?out.+?of.*?',
+  },
+  {
+    key: 'Processed messages out of',
+    examples: ['Processed out 676 messages messages out of 676 Processed'],
+    regex: '.*?Processed.+?messages.+?out.+?of.*?',
+  },
+  {
+    key: 'Processed of',
+    examples: ['Processed 676 messages out of 676'],
+    regex: '.*?Processed.+?of.*?',
+  },
+  {
+    key: 'Processed messages out of',
+    examples: ['Processed messages out of'],
+    regex: '.*?Processed.+?messages.+?out.+?of.*?',
+  },
+  {
+    key: '',
+    examples: ['Processed messages out of'],
+    regex: '.*?',
+  },
+  {
+    key: 'Processed messages out of',
+    examples: ['Processed 676 (*?) message* out of 676'],
+    regex: '.*?Processed.+?messages.+?out.+?of.*?',
+  },
+  {
+    key: '2024 0000 admin to admin console.prod 6000 api HTTP 1.1 https admin Mozilla 5.0 AppleWebKit KHTML like Gecko Chrome Safari',
+    examples: [
+      '[05/Jan/2024 04:11:42 +0000] 40.69.144.53 - Lucio77 admin-console.you-got.mail to: admin-console.prod.008:6000: "GET /api/listCustomers HTTP/1.1" 200 383 "https://admin-console.you-got.mail" "Mozilla/5.0 (Windows; U; Windows NT 6.3) AppleWebKit/531.1.1 (KHTML, like Gecko) Chrome/23.0.835.0 Safari/531.1.1"',
+    ],
+    regex:
+      '.*?2024.+?0000.+?admin.+?to.+?admin.+?console.prod.+?6000.+?api.+?HTTP.+?1.1.+?https.+?admin.+?Mozilla.+?5.0.+?AppleWebKit.+?KHTML.+?like.+?Gecko.+?Chrome.+?Safari.*?',
+  },
+] as Category[];
+
+describe('FormattedPatternExamples', () => {
+  it('renders each category correctly', () => {
+    categories.forEach((category) => {
+      expect(<FormattedPatternExamples category={category} />).toMatchSnapshot();
+      expect(<FormattedRegex category={category} />).toMatchSnapshot();
+      expect(<FormattedTokens category={category} />).toMatchSnapshot();
```

Comment:
I'd like to suggest that we try to avoid relying on snapshot testing for new unit tests. A while ago we also documented old tests that we wanted to refactor and no longer rely on snapshot tests (https://github.com/elastic/kibana/issues/153288). The snapshots render the component with its props but do not assert what's actually rendered.

An alternative with `react-testing-lib` would look something like:

```ts
import {render, screen} from '@testing-library/react';
import '@testing-library/jest-dom';

...

describe('Format category components', () => {
    it('renders FormattedPatternExamples correctly', async () => {
      render(<FormattedPatternExamples category={category} />);
      expect(screen.getByText(/some text we expect to be rendered/i)).toBeInTheDocument();
    });
    ...
});
```

You could also adapt this to assert just certain pattern elements you'd expect to be rendered within `span` elements for the custom highlighting.

Reply:
All I'm really interested here is the number of elements created, as in, how the line has been split up. 
My original version of this test did just that, I'd count the elements and ensure that the line had been split up correctly.
Then I noticed that I'd get the same result by using snapshot.
I can revert this to counting the elements. Shame I deleted the code.

Reply:
I've updated the test in https://github.com/elastic/kibana/pull/175320/commits/ff9785b444d7e864589dc7fda7b8fef3b7be39b1
Reverting it back to a test to count the number of elements returned from `createFormattedExample`